content/doc/book/pipeline/shared-libraries.adoc

```diff
@@ -412,57 +412,52 @@ node {
 === Defining global variables
 
 Internally, scripts in the `vars` directory are instantiated on-demand  as
-singletons. This allows multiple methods or properties to be defined in a
-single `.groovy` file which interact with each other, for example:
+singletons. This allows multiple methods to be defined in a
+single `.groovy` file for convenience.  For example:
 
+.var/log.groovy
 [source,groovy]
 ----
-// vars/acme.groovy
-def setName(value) {
-    name = value
+def info(message) {
+    echo "INFO: ${message}"
 }
-def getName() {
-    name
-}
-def caution(message) {
-    echo "Hello, ${name}! CAUTION: ${message}"
+
+def warning(message) {
+    echo "WARNING: ${message}"
 }
 ----
 
-In the above, `name` is not referring to a field (even if you write it as `this.name`!),
-but to an entry created on demand in a `Script.binding`.
-To be clear about what data you intend to store and of what type,
-you can instead provide an explicit class declaration
-(the class name should match the file basename, and Pipeline steps can
-only be invoked if `steps` or `this` is passed to the class or method,
-as with classes in `src` described above):
-
+.Jenkinsfile
 [source,groovy]
 ----
-// vars/acme.groovy
-class acme implements Serializable {
-    private String name
-    def setName(value) {
-        name = value
-    }
-    def getName() {
-        name
-    }
-    def caution(message) {
-        echo "Hello, ${name}! CAUTION: ${message}"
-    }
-}
+@Library('utils') _
+
+log.info 'Starting'
+log.warning 'Nothing to do!'
 ----
 
-The Pipeline can then invoke these methods which will be defined on the
-`acme` object:
+Declarative Pipeline does not allow global variable usage outside of a `script`
+directive
+(link:https://issues.jenkins-ci.org/browse/JENKINS-42360[JENKINS-42360]).
```

Comment:
@abayer correct me if I am wrong, but I think so long as you declare the method as `call`, you can call it as if it were a step.

Reply:
@jglick I understand that from one perspective everything defined under `vars` is a "global variable.  However, from a user perspective, "global variables" defined with the`call`-method  do not behave the same way as "global variables" defined with other method names.

Perhaps we could agree to call them "global variables" and "global functions"?

Reply:
Any variable referring to an object with a `call` method can be called like a function. This is just Groovy sugar. Technically Groovy has no “functions” at all. It has methods, etc.